myhdl/conversion/_VHDLNameValidation.py

```diff
@@ -0,0 +1,47 @@
+import warnings
+from myhdl import ToVHDLWarning
+
+class _nameCheck():
+    'Saves all reserved words in VHDL and variable names used in a MyHDL circuit to check for any name collisions'
+
+    #A list of all reserved words within VHDL which should not be used for
+    #anything other than their own specific purpose
+    _vhdl_keywords = ["abs", "access", "after", "alias", "all",
+                  "and", "architecture", "array", "assert",
+                  "attribute", "begin", "block", "body", "buffer",
+                  "bus", "case", "component", "configuration",
+                  "constant", "disconnect", "downto", "else",
+                  "elseif", "end", "entity", "exit", "file", "for",
+                  "function", "generate", "generic", "group",
+                  "guarded", "if", "impure", "in", "inertial",
+                  "inout", "is", "label", "library", "linkage",
+                  "literal", "loop", "map", "mod", "nand", "new",
+                  "next", "nor", "not", "null", "of", "on", "open",
+                  "or", "others", "out", "package", "port",
+                  "postponed", "procedure", "process", "pure",
+                  "range", "record", "register", "reject", "rem",
+                  "report", "return", "rol", "ror", "select",
+                  "severity", "signal", "shared", "sla", "sll", "sra",
+                  "srl", "subtype", "then", "to", "transport", "type",
+                  "unaffected", "units", "until", "use", "variable",
+                  "wait", "when", "while", "with", "xnor", "xor"];
+
+    #A list to hold all signal names being used in lowercase to raise an error
+    #if no names are reused with different casing
+    _usedNames = [];
+
+    #Function which compares current parsed signal/entity to all keywords to
+    #ensure reserved words are not being used for the wrong purpose
+    def _nameValid(keyword):
+        for i in _nameCheck._vhdl_keywords:
```

Comment:
This code is incorrect, it should be something like:

```python
def _nameValid(name):
    for keyword in _nameCheck._vhdl_keywords:
        if name == keyword:
```

The value passed to the checking function is not the keyword but rather the parsed strings/names.